### PR TITLE
gh-87901: os.popen: Remove the encoding argument

### PR DESCRIPTION
**Pull request by methane**

### CODE REVIEW

**Comment by vstinner** [comment]
Since the parameter was added to Python 3.11, I guess your intent is to backport this change to 3.11.

What's the rationale for adding the parameter and then to remove it? https://github.com/python/cpython/issues/87901 doesn't give details.

**Comment by methane** [comment]
I want to fix EncodingWarning in stdlib as possible.

Adding `encoding` parameter is the straight way to fix EncodingWarning. User can pass encoding explicitly.

This PR doesn't fix the EncodingWarning in `os.popen`. It navigate users to migrate to `subprocess.Popen` if they want to use consistent encoding regardless UTF-8 mode.

**Comment by vstinner** [comment]
Since I wish I could deprecate os.popen(), I like the idea of suggesting users to use the subprocess module directly. It's just that I was surprised to see the parameter added and then removed. Now I understand the rationale, thanks.

File: Doc/library/os.rst [hunk 1]
```diff
@@ -3945,8 +3945,14 @@ written in Python, such as a mail server's external command delivery program.
...
-   .. versionchanged:: 3.11
-      Added the *encoding* parameter.
+   .. note::
+      On Unix, the *cmd* argument is encoded with
+      :term:`file system encoding <filesystem encoding and error handler>`.
+      The returned file object connected to the pipe uses
+      :func:`locale.getpreferredencoding(False) <locale.getpreferredencoding>`_.
+      The :ref:`Python UTF-8 Mode <utf8-mode>` affects both encodings.
+
+      Use :class:`subprocess.Popen` to control encoding and many other options.
```
Comment (vstinner) [inline Doc/library/os.rst]: The last sentence "... use subprocess to control encoding ..." is fine. But I'm not sure that it's worth it to maintain a documentation of the encoding used by subprocess here. I'm not against it ;-)

**Review by vstinner — APPROVED** [review]
LGTM. I just added a remark on the doc.

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**